src/osp/Active/SysVehicle.cpp

```diff
@@ -261,11 +261,31 @@ void SysVehicle::part_instantiate_machines(ActiveScene& rScene, ActiveEnt partEn
     }
 }
 
+float SysVehicle::compute_hier_volume(ActiveScene& rScene, ActiveEnt part)
+{
+    float volume = 0.0f;
+    auto checkVol = [&rScene, &volume](ActiveEnt ent)
+    {
+        auto const* shape = rScene.get_registry().try_get<ACompShape>(ent);
+        if (shape)
+        {
+            auto const& xform = rScene.reg_get<ACompTransform>(ent);
+            volume += shape_volume(shape->m_shape, xform.m_transform.scaling());
+        }
+        return EHierarchyTraverseStatus::Continue;
```

Comment:
I suspect that the compiler *should* be detecting that this lambda always returns the same value unconditionally,

but if you wanted to be super conservative, we could improve hierarchy_traverse to check the return type of the lambda, and for lambdas that return void, instead of trying to use the return type, it assumes that the lambda would have returned ::Continue.

You'd do something like

template<typename FUNCTOR_T, bool ALWAYS_CONTINUE = std::is_same_t<void, std::result_of_t<FUNCTOR_T(ActiveEnt)>>>
blah foo(FUNCTOR_T)
{
    if constexpr(ALWAYS_CONTINUE)
    {
        // simple version
    }
    else
    {
        // complicated version
    }
}

Reply:
Or you could use partial template specialization.

```
template<typename FUNCTOR_T,
         bool ALWAYS_CONTINUE = std::is_void_t<std::result_of_t<FUNCTOR_T(ActiveEnt)>>>
blah foo(FUNCTOR_T)
{
    static_assert("Well that's not supposed to happen...");
}

template<typename FUNCTOR_T>
blah foo<FUNCTOR_T, true>(FUNCTOR_T)
{
    // simple version
}

template<typename FUNCTOR_T>
blah foo<FUNCTOR_T, false>(FUNCTOR_T)
{
    // complex version version
}
```